app/controllers/concerns/grouping_helper.rb

```diff
@@ -0,0 +1,70 @@
+module GroupingHelper
+
+  class GroupedIncumbency
+    attr_accessor :start_date, :end_date, :nodes, :grouped_by, :type
+  end
+
+  # Returns array of grouped, ungrouped and unknown objects
+  def self.group(data, category, property)
+    grouped_data = group_data(data, category, property)
+    create_sorted_array(grouped_data)
+  end
+
+  # Returns hash with grouped keys
+  # e.g. { 12345678 => [...], 23456789 => [...], "UNKNOWN" => [...] }
+  def self.group_data(data, category, property)
+    data.group_by { |data_point| data_point.try(category).try(property) || 'UNKNOWN' }
+  end
+
+  def self.separate_unknown_and_ungrouped_data(value)
+    [] << value
+  end
+
+  def self.separate_grouped_data(data_hash, key)
+    grouped = []
+
+    grouped_object = GroupingHelper::GroupedIncumbency.new
+    grouped_object.grouped_by = key
+    grouped_object.nodes = data_hash[key]
+
+    start_date = nil
+    end_date = nil
+    current_node = nil
+
+    grouped_object.nodes.each do |node|
+      # Find node that represents current role
+      current_node = node if node.try(:end_date).nil? || node.end_date.nil?
+      # Find earliest start date and latest end date
+      start_date = node.start_date if start_date.nil? || node.start_date < start_date
+      end_date = node.end_date if !current_node && (end_date.nil? || node.end_date > end_date)
+    end
+
+    # Set properties of the object
+    grouped_object.type = grouped_object.nodes.first.type
```

Comment:
maybe add a todo comment to check if this is still okay when we have roles and memberships of a committee